### DISCUSSION TITLE
Negating duals in Hessian

### DISCUSSION THREAD

**Discussion by tso-martin** [mkhraijah/PowerModelsADA.jl#15] [General]
Dear @mkhraijah,
I wonder why the duals of all constraints are negated in https://github.com/mkhraijah/PowerModelsADA.jl/blob/main/src/algorithms/aladin_coordinated_methods.jl#L403. Is there a special reason for this?
Best regards,
Martin

**Comment by mkhraijah** [comment]
Hi @tso-martin,
This implementation follows the heuristic in Section A3 of the paper: [ALADIN-$`\alpha`$—An open-source MATLAB toolbox for distributed non-convex optimization](https://onlinelibrary.wiley.com/doi/full/10.1002/oca.2811). We negate the dual variables to account for the JuMP convention on the sign of dual variables. We also negate any negative eigenvalue to ensure the Hessian matrix is positive-semidefinite. In any case, we are working on rewriting the ALADIN code using the new nonlinear features in JuMP. So, this part will be changed. 
Regards, 
Mohannad
